lib/puppet/type/cisco_acl.rb

```diff
@@ -0,0 +1,103 @@
+#
+# Puppet resource type for acl
+#
+# Copyright (c) 2014-2015 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+Puppet::Type.newtype(:cisco_acl) do
+  # ---------------------------------------------------------------
+  # STEP 1. Create a @doc entry to describe the resource and usage
+  # ---------------------------------------------------------------
+  @doc = "Manages configuration of a acl instance
+
+  ```
+  cisco_acl {'<string>':
+    ..attributes..
+  }
+  ```
+
+  `<string>` is the name of the acl instance.
+
+  Example:
+
+  ```
+    cisco_acl { 'ipv4 foo' :
+      ensure => present,
+      stats_per_entry => false,
+      fragments => 'permit'
+    }
+
+  ```
```

Comment:
This type needs to support 3 types of title patterns, and needs puppet formatting, so update this to:

```
\~~~puppet
    cisco_acl { 'ipv4 my_acl' :
      ensure               => present,
      stats_per_entry => false,
      fragments          => 'permit'
    }
~~~

Example Title Patterns

\~~~puppet
    cisco_acl { 'new_york' :
      ensure               => present,
      afi                     => 'ipv4',
      acl_name          => 'my_acl'
    }
~~~

\~~~puppet
    cisco_acl { 'ipv4' :
      ensure               => present,
      acl_name          => 'my_acl'
    }
~~~

\~~~puppet
    cisco_acl { 'ipv4 my_acl' :
      ensure               => present,
    }
~~~
```